modules/xfeatures2d/test/test_features2d.cpp

```diff
@@ -85,6 +85,12 @@ TEST( Features2d_Detector_Harris_Laplace_Affine, regression )
     test.safe_run();
 }
 
+TEST(Features2d_Detector_TBMR_Affine, regression)
+{
+    CV_FeatureDetectorTest test("detector-tbmr-affine", TBMR::create());
```

Comment:
Please prepare PR with test data into `opencv_extra` repository (using the same source branch name).
To receive accurate result please use:
- "Debug" build with disabled optimizations: `cmake -DCV_DISABLE_OPTIMIZATION=ON ...` (please run this in new (or clean) build directory, because CMake caches some values like `WITH_IPP`).
- disable multi-threading: `./bin/opencv_test_xfeatures2d --gtest_filter=*TBMR* --test_threads=1`

Reply:
with a PR in opencv_extra as well?

Reply:
Yes, please open PR in `opencv_extra` for reviewing and merging.

Reply:
ok there it is: [TBMR features (testdata)](https://github.com/opencv/opencv_extra/pull/817)